OpenRA.Mods.Common/Traits/Buildings/ActorPreviewPlaceBuildingPreview.cs

```diff
@@ -0,0 +1,102 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Mods.Common.Orders;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Creates a building placement preview based on the map editor actor preview.")]
+	public class ActorPreviewPlaceBuildingPreviewInfo : FootprintPlaceBuildingPreviewInfo
+	{
+		[Desc("Enable the building's idle animation")]
+		public readonly bool Animated = true;
+
+		[PaletteReference("OverridePaletteIsPlayerPalette")]
+		[Desc("Custom palette name")]
+		public readonly string OverridePalette = null;
+
+		[Desc("Custom palette is a player palette BaseName")]
+		public readonly bool OverridePaletteIsPlayerPalette = true;
+
+		protected override IPlaceBuildingPreview CreatePreview(WorldRenderer wr, ActorInfo ai, TypeDictionary init)
+		{
+			return new ActorPreviewPlaceBuildingPreviewPreview(wr, ai, this, init);
+		}
+
+		public override object Create(ActorInitializer init)
+		{
+			return new ActorPreviewPlaceBuildingPreview();
+		}
+	}
+
+	public class ActorPreviewPlaceBuildingPreview { }
+
+	class ActorPreviewPlaceBuildingPreviewPreview : FootprintPlaceBuildingPreviewPreview
```

Comment:
I am aware of how ridiculous this name is, but it comes logically from the naming scheme - it is the concrete preview object for the PlaceBuildingPreview, that happens to be drawn using the ActorPreview...

I'm open for other options.